Tools/cases_generator/generate_cases.py

```diff
@@ -183,17 +181,15 @@ def effect_str(effects: list[StackEffect]) -> str:
                     assert target_instr
                     target_popped = effect_str(target_instr.input_effects)
                     target_pushed = effect_str(target_instr.output_effects)
-                    if popped is None and pushed is None:
-                        popped, pushed = target_popped, target_pushed
-                    else:
-                        assert popped == target_popped
-                        assert pushed == target_pushed
+                    popped, pushed = target_popped, target_pushed
```

Comment:
Those asserts need to stay -- per the comment on line 174-175, we're checking here that all targets of the pseudo have the same effects.